mne/simulation/metrics/metrics.py

```diff
@@ -0,0 +1,572 @@
+    use term Region Localization Error (RLE).
+
+    .. versionadded:: 1.2
+
+    References
+    ----------
+    .. footbibliography::
+    """  # noqa: E501
+    stc_true, stc_est = _uniform_stc(stc_true, stc_est)
+    stc_true, stc_est = _thresholding(stc_true, stc_est, threshold)
+    func = partial(_dle, src=src, stc=stc_true)
+    metric = _apply(func, stc_true, stc_est, per_sample=per_sample)
+    return metric
+
+
+def _roc_auc_score(p, q):
+    from sklearn.metrics import roc_auc_score
+    return roc_auc_score(np.abs(p) > 0, np.abs(q))
+
+
+@fill_doc
+def roc_auc_score(stc_true, stc_est, per_sample=True):
+    """Compute ROC AUC between 2 source estimates.
+
+    ROC stands for receiver operating curve and AUC is Area under the curve.
+    When computing this metric the stc_true must be thresholded
+    as any non-zero value will be considered as a positive.
+
+    The ROC-AUC metric is computed between amplitudes of the source
+    estimates, i.e. after taking the absolute values.
+
+    Parameters
+    ----------
+    %(stc_true_metric)s
+    %(stc_est_metric)s
+    %(per_sample_metric)s
+
+    Returns
+    -------
+    %(stc_metric)s
+
+    Notes
+    -----
+    .. versionadded:: 1.2
+    """
+    stc_true, stc_est = _uniform_stc(stc_true, stc_est)
+    func = partial(_roc_auc_score)
+    metric = _apply(func, stc_true, stc_est, per_sample=per_sample)
+    return metric
+
+
+def _f1_score(p, q):
+    from sklearn.metrics import f1_score
+    p = np.sum(np.abs(p), axis=1)
+    q = np.sum(np.abs(q), axis=1)
+    return f1_score(p > 0, q > 0)
+
+
+@fill_doc
+def f1_score(stc_true, stc_est, threshold='90%', per_sample=True):
+    """Compute the F1 score, also known as balanced F-score or F-measure.
+
+    The F1 score can be interpreted as a weighted average of the precision
+    and recall, where an F1 score reaches its best value at 1 and worst score
+    at 0. The relative contribution of precision and recall to the F1
+    score are equal.
+    The formula for the F1 score is::
+
+        F1 = 2 * (precision * recall) / (precision + recall)
+
+    Threshold is used first for data binarization.
+
+    Parameters
+    ----------
+    %(stc_true_metric)s
+    %(stc_est_metric)s
+    threshold : float | str
+        The threshold to apply to source estimates before computing
+        the f1 score. If a string the threshold is
+        a percentage and it should end with the percent character.
+    %(per_sample_metric)s
+
+    Returns
+    -------
+    %(stc_metric)s
+
+    Notes
+    -----
+    .. versionadded:: 1.2
+    """
+    stc_true, stc_est = _uniform_stc(stc_true, stc_est)
+    stc_true, stc_est = _thresholding(stc_true, stc_est, threshold)
+    func = partial(_f1_score)
+    metric = _apply(func, stc_true, stc_est, per_sample=per_sample)
+    return metric
+
+
+def _precision_score(p, q):
+    from sklearn.metrics import precision_score
+    p = np.sum(np.abs(p), axis=1)
+    q = np.sum(np.abs(q), axis=1)
```

Comment:
these 2 lines are copied many times for all private functions. This should be refactored to avoid this